ui/desktop/src/components/SessionListView.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useEffect, useState } from 'react';
+import { ViewConfig } from '../App';
+import { MessageSquare, Loader, AlertCircle, Calendar, ChevronRight } from 'lucide-react';
+import { fetchSessions, type Session } from '../sessions';
+import { Card } from './ui/card';
+import { Button } from './ui/button';
+import BackButton from './ui/BackButton';
+import { ScrollArea } from './ui/scroll-area';
+
+interface SessionListViewProps {
+  setView: (view: ViewConfig['view'], viewOptions?: Record<any, any>) => void;
+  onSelectSession: (sessionId: string) => void;
+}
+
+const SessionListView: React.FC<SessionListViewProps> = ({ setView, onSelectSession }) => {
+  const [sessions, setSessions] = useState<Session[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    // Load sessions on component mount
+    loadSessions();
+  }, []);
+
+  const loadSessions = async () => {
+    setIsLoading(true);
+    setError(null);
+    try {
+      const response = await fetchSessions();
+      setSessions(response.sessions);
+    } catch (err) {
+      console.error('Failed to load sessions:', err);
+      setError('Failed to load sessions. Please try again later.');
+      setSessions([]);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  // Format date to be more readable
+  // eg. "10:39 PM, Feb 28, 2025"
+  const formatDate = (dateString: string) => {
+    try {
+      const date = new Date(dateString);
+      const time = new Intl.DateTimeFormat('en-US', {
+        hour: 'numeric',
+        minute: 'numeric',
+        hour12: true,
+      }).format(date);
+
+      const dateStr = new Intl.DateTimeFormat('en-US', {
+        month: 'short',
+        day: 'numeric',
+        year: 'numeric',
+      }).format(date);
+
+      return `${time}, ${dateStr}`;
+    } catch (e) {
+      return dateString;
+    }
+  };
+
+  return (
+    <div className="h-screen w-full">
+      <div className="relative flex items-center h-[36px] w-full bg-bgSubtle"></div>
+
+      <ScrollArea className="h-full w-full">
+        <div className="flex flex-col pb-24">
+          <div className="px-8 pt-6 pb-4">
+            <BackButton onClick={() => setView('chat')} />
+          </div>
+
+          {/* Content Area */}
+          <div className="flex flex-col mb-6 px-8">
+            <h1 className="text-3xl font-medium text-textStandard">Previous goose sessions</h1>
+            <h3 className="text-sm text-textSubtle mt-2">
+              View previous goose threads and their contents to pick up where you left off.
```

Comment:
Nit: use "goose sessions" for consistency as all other things I see call them sessions vs threads